cf-agent/verify_users_pam.c

```diff
@@ -994,7 +994,8 @@ static bool VerifyIfUserNeedsModifs (const char *puser, User u, const struct pas
             CFUSR_SETBIT (*changemap, i_group);
         }
     }
-    if (u.groups_secondary != NULL)
+
+    if (u.groups_secondary_given)
```

Comment:
It should remove you from all groups if `groups_secondary` is empty, and just preserve existing membership if it's missing. At least this was the intention. Right now I don't recall if this is tested. So yeah, this makes sense to me.